### PR TITLE
Adding support for self signed certificate

### PR DESCRIPTION
**Pull request by Bouk250**
### Checklist for PRs

- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Documented code changes according to the [CSI standard](https://standards.mousepawmedia.com/en/stable/csi.html)
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs and all suitable `__all__` s

### CODE REVIEW

**Comment by Bibo-Joshi** [comment]
Hi. can you explain your use case for this?

**Comment by Bouk250** [comment]
Hi, yes.

My network connection pass over gateway with self signed ssl certificate.
The problem is when i try to set up bot i gon an exception:
SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1131)')

So I added a verify function to the application builder for adding support to ssl certfile. with pem file path ...

**Comment by Bibo-Joshi** [comment]
> My network connection pass over gateway with self signed ssl certificate. 

Sorry, I don't quite understand. what kind of gateway? I'm asking because I'd like to evaluate how specific to your use case this issue is, i.e. if the use case is broad enough to add it to the library :)

**Comment by Poolitzer** [comment]
Im sorry what is happening here. I used https://docs.python-telegram-bot.org/en/v20.0a0/telegram.ext.application.html#telegram.ext.Application.run_webhook with the cert and key option + self signed certificate just fine before the alpha release. Is there a traceback happening when you use that?

**Comment by Poolitzer** [comment]
@Bibo-Joshi ohh the git diff hid that from my eyes, im sorry

**Comment by Bouk250** [comment]
> Sorry, I don't quite understand. what kind of gateway? I'm asking because I'd like to evaluate how specific to your use case this issue is, i.e. if the use case is broad enough to add it to the library :)

For example it's can happen when the bot connection pass over company network, sub-network, proxy...

File: telegram/request/_httpxrequest.py [hunk 1]
```diff
@@ -105,10 +108,13 @@ def __init__(
             max_connections=connection_pool_size,
             max_keepalive_connections=connection_pool_size,
         )
+
```
Comment (Bibo-Joshi) [inline telegram/request/_httpxrequest.py:111]: please remove

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
> For example it's can happen when the bot connection pass over company network, sub-network, proxy...

Okay, that sounds somewhat reasonable to me and I'd be willing to accept a contribution for this.

I left a number of review comments below. In addition, please make sure to cover all applicable items of the pr-check list + everything mentioned in the [contribution guide](https://docs.python-telegram-bot.org/en/v20.0a1/contributing.html). If you're having troubles with adding & adapting unit tests, feel free to reach out here or in the [dev group](https://t.me/pythontelegrambotchannel/105).

**Comment by Bibo-Joshi** [comment]
I'll close this due to inactivity for now. If you'd like to pick up this PR, feel free to comment/open a new pr :)

**Pull request closed without merging**